lib/index.js

```diff
@@ -17,183 +17,174 @@ internals.defaults = {
     port: 6379
 };
 
+module.exports = class Connection {
+    constructor(options) {
 
-exports = module.exports = internals.Connection = function (options) {
+        assert(this.constructor === Connection, 'Redis cache client must be instantiated using new');
 
-    Hoek.assert(this.constructor === internals.Connection, 'Redis cache client must be instantiated using new');
-
-    this.settings = Object.assign({}, internals.defaults, options);
-    this.client = null;
-    return this;
-};
-
-
-// Async
-internals.Connection.prototype.start = function () {
-
-    if (this.settings.client) {
-        this.client = this.settings.client;
+        this.settings = Object.assign({}, internals.defaults, options);
+        this.client = null;
+        return this;
     }
 
-    if (this.client) {
-        return Promise.resolve();
-    }
+    async start() {
 
-    // Return a promise that is resolved when everything is ready
-    return new Promise((resolve, reject) => {
+        if (this.settings.client) {
+            this.client = this.settings.client;
+        }
 
-        let client;
+        if (this.client) {
+            return;
+        }
 
-        const options = {
-            password: this.settings.password,
-            db: this.settings.database || this.settings.db,
-            tls: this.settings.tls
-        };
+        // Return a promise that is resolved when everything is ready
+        return await new Promise((resolve, reject) => {
```

Comment:
Why did you add the `await` in `return await new Promise(…)`?
I thought we can leave it out and the code behaves the same. Does it?

Reply:
lab lint complains otherwise.